main/webservices/api/test_api.php

```diff
@@ -2,116 +2,564 @@
+        $extraFieldValue = $extraFieldValueModel->get_values_by_handler_and_field_variable(
+            $newSessionId,
+            $EXTRA_FIELD_NAME);
+        $this->assertNotFalse($extraFieldValue);
+        $this->assertSame($EXTRA_FIELD_VALUE_FOR_NEW_SESSION, $extraFieldValue['value']);
+
+        // assert the model session still has its own original value in the first extra field
+        $extraFieldValue = $extraFieldValueModel->get_values_by_handler_and_field_variable(
+            $modelSessionId,
+            $EXTRA_FIELD_NAME);
+        $this->assertNotFalse($extraFieldValue);
+        $this->assertSame($EXTRA_FIELD_VALUE_FOR_MODEL_SESSION, $extraFieldValue['value']);
+
+        // assert the new session has inherited the model session value in the second extra field
+        $extraFieldValue = $extraFieldValueModel->get_values_by_handler_and_field_variable(
+            $newSessionId,
+            $SECOND_EXTRA_FIELD_NAME);
+        $this->assertNotFalse($extraFieldValue);
+        $this->assertSame($SECOND_EXTRA_FIELD_VALUE, $extraFieldValue['value']);
 
-    if ($response->getStatusCode() !== 200) {
-        throw new Exception('Entry denied with code : ' . $response->getStatusCode());
+        // assert the model session still has the same value in the second extra field
+        $extraFieldValue = $extraFieldValueModel->get_values_by_handler_and_field_variable(
+            $modelSessionId,
+            $SECOND_EXTRA_FIELD_NAME);
+        $this->assertNotFalse($extraFieldValue);
+        $this->assertSame($SECOND_EXTRA_FIELD_VALUE, $extraFieldValue['value']);
+
+        // assert the new session inherited the model session courses
+        $modelCourseList = array_keys(SessionManager::get_course_list_by_session_id($modelSessionId));
+        $newCourseList = array_keys(SessionManager::get_course_list_by_session_id($newSessionId));
+        $this->assertSame($modelCourseList, $newCourseList);
+
+        // assert the current url was set on the new session
+        if (api_is_multiple_url_enabled()) {
+            $urls = UrlManager::get_access_url_from_session($newSessionId);
+            $this->assertSame(1, count($urls));
+            $this->assertSame(api_get_current_access_url_id(), intval($urls[0]['access_url_id']));
+        }
+
+        // clean up
+        foreach ($COURSE_CODES as $code) {
+            CourseManager::delete_course($code);
+        }
+        SessionManager::delete($modelSessionId);
+        SessionManager::delete($newSessionId);
+        $promotion->delete($promotionId);
+        $career->delete($careerId);
+        $extraFieldModel->delete($extraFieldId);
+        $extraFieldModel->delete($secondExtraFieldId);
+        if (api_is_multiple_url_enabled()) {
+            UrlManager::delete($urlId);
+        }
     }
 
-    $jsonResponse = json_decode($response->getBody()->getContents());
+    public function testSubscribeUserToSessionFromUsername()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $ACTION is not named in camelCase.](https://app.codacy.com/app/chamilo/chamilo-lms/pullRequest?prid=4554450)